### ISSUE TITLE
via lines often result in git conflict

### ISSUE THREAD

**Issue by graingert** [jazzband/pip-tools#1036]
#### What's the problem this feature will solve?
currently when adding and removing packages in multiple github PRs the `# via` line will conflict

#### Describe the solution you'd like
put each `# via` on a new line:
```
six==1.12.0 \
    --hash=sha256:3350809f0555b11f552448330d0b52d5f24c91a322ea4a15ef22629740f3761c \
    --hash=sha256:d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73 \
    # via automat, bcrypt, cryptography, ecdsa, format-cef, grpcio, patchy, pathlib, prompt-toolkit, protobuf, pyhamcrest
```
instead use:
```
six==1.12.0 \
    --hash=sha256:3350809f0555b11f552448330d0b52d5f24c91a322ea4a15ef22629740f3761c \
    --hash=sha256:d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73 \
    # via automat
    #   , bcrypt
    #   , cryptography
    #   , ecdsa
    #   , format-cef
    #   , grpcio
    #   , patchy
    #   , pathlib
    #   , prompt-toolkit
    #   , protobuf
    #   , pyhamcrest
```

**Comment by graingert** [comment]
also the missing trailing slash on deps without `# via` lines also conflict - this would be fixed with https://github.com/jazzband/pip-tools/issues/881

**Comment by jdufresne** [comment]
I've been running into the same frustration.

This is especially noticeable on Django projects that include many Django 3rd party libraries. The Django "via" line can easily extend beyond column 200.

Proposed fix in #1237. If there is a single annotation, then it is rendered on one line, otherwise, one per line.

**Comment by graingert** [comment]
> also the missing trailing slash on deps without `# via` lines also conflict - this would be fixed with #881

@jdufresne it looks like there's now always a missing trailing slash - this increases the chance of a merge conflict

**Comment by jdufresne** [comment]
The final trailing slash doesn't make sense. The slash indicates "escape this newline, to continue the statement".

I agree that it is unfortunate that this can lead to merge conflicts, but it is still wrong placement, IMO.

For example, the text:

```
six==1.12.0 \
    --hash=sha256:3350809f0555b11f552448330d0b52d5f24c91a322ea4a15ef22629740f3761c \
    --hash=sha256:d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73
    # via
    #   bcrypt
    #   cryptography
```

After expanding the newline escapes, should expand to:

```
six==1.12.0 --hash=sha256:3350809f0555b11f552448330d0b52d5f24c91a322ea4a15ef22629740f3761c --hash=sha256:d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73
    # via
    #   bcrypt
    #   cryptography
```

Notice the first comment is on the next line.

It should not expand to:

```
six==1.12.0 --hash=sha256:3350809f0555b11f552448330d0b52d5f24c91a322ea4a15ef22629740f3761c --hash=sha256:d16a0141ec1a18405cd4ce8b4613101da75da0e9a7aec5bdd4fa804d0e0eba73  # via
    #   bcrypt
    #   cryptography
```

The final newline happens to work as the next line is a comment, but if someone disabled annotations (`--no-annotate`), the next line would no longer be a comment.
